From cf75179da42dc2b63321b894f35f4695ccaebbd3 Mon Sep 17 00:00:00 2001
From: Fedor Pchelkin <pchelkin@ispras.ru>
Date: Thu, 23 Nov 2023 16:05:12 +0300
Subject: [PATCH] Implement PatchDiff: patch2: handle maze symbols and maze size

---
 09_PatchDiff/maze-1.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/09_PatchDiff/maze-1.c b/09_PatchDiff/maze-1.c
index 8445768..01ad947 100644
--- a/09_PatchDiff/maze-1.c
+++ b/09_PatchDiff/maze-1.c
@@ -1,5 +1,7 @@
+#include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <time.h>
 
 typedef struct {
@@ -112,8 +114,18 @@ void freeMaze(Maze *maze)
 
 int main(int argc, char *argv[])
 {
-	int size = 6;
-	Maze maze = {size, size * 2 + 1, '.', '#', NULL, NULL, 0};
+	if (argc != 3) {
+		errno = EINVAL;
+		panic("Provide maze symbols and maze size");
+	}
+	if (strlen(argv[1]) < 2) {
+		errno = EINVAL;
+		panic("Provide at least two maze symbols");
+	}
+	char pass_symb = argv[1][0];
+	char wall_symb = argv[1][1];
+	int size = strtol(argv[2], NULL, 10);
+	Maze maze = {size, size * 2 + 1, pass_symb, wall_symb, NULL, NULL, 0};
 
 	srand(time(NULL));
 	initializeMaze(&maze);
-- 
2.34.1

